src/creature_states.c

```diff
@@ -4220,6 +4220,10 @@ TbBool cleanup_creature_state_and_interactions(struct Thing *creatng)
 
 TbBool can_change_from_state_to(const struct Thing *thing, CrtrStateId curr_state, CrtrStateId next_state)
 {
+    if (is_thing_directly_controlled_by_player(thing, my_player_number))
```

Comment:
I see you specifically check if it's possessed by the specifically by the my_player instead of any player, wouldn't that cause a desync if it happens to another player in multiplayer? didn't test anything, just doesn't sound logical

Reply:
I already commented this on the discord. I think it will also interrupt with stuff like 'changing into chicken' if the creature in possession is targeted by enemy keeper.

Reply:
> I see you specifically check if it's possessed by the specifically by the my_player instead of any player, wouldn't that cause a desync if it happens to another player in multiplayer? didn't test anything, just doesn't sound logical

I don't know. If someone could test that, it would be good. There was code that prevented a creature from being drawn if it's controlled (which is why you can't see the creature you're possessing), but it only checked the creature's owner. It also sounded to me as if a creature who's being possessed would disappear for everyone in multiplayer, which is far from ideal.

Reply:
for rendering it makes sense to differentiate between my_player and other players, for changing states however all states should remain in sync regardless of on who's computer the logic runs right?